### PR TITLE
Enhancement/mvp 176 standardize tag prefix scan searches

### PR DESCRIPTION
**Pull request by eduwercamacaro**
First steps for improve Search mechanism.

`PublicScanRequest` types usually do this logic on tag scan:

1. Find if the search should be Local or Remote
2. Set storage name (Core or repartition)
3. Build attributes from proto Search class
4. Create tag prefix from attributes
5. handle bookmark and timestamps.

The intention for this change is to improve step 3 and 4 in a generic way.

### CODE REVIEW

**Review by sauljabin — APPROVED** [review]

File: schemas/internal_server.proto [hunk 1]
```diff
@@ -85,9 +85,9 @@ message BookmarkPb { // Used for paginated responses
 
 message InternalScanPb {
     message TagScanPb {
-        repeated AttributePb attributes = 1;
-        optional google.protobuf.Timestamp earliest_create_time = 2;
-        optional google.protobuf.Timestamp latest_create_time = 3;
+        optional google.protobuf.Timestamp earliest_create_time = 1;
```
Comment (MateoRojas) [inline schemas/internal_server.proto:88]: This is fine for now, but after we release for Elevance we would need to ensure compatibility. 

P.S. Tech Huddle question: How can we ensure that we have automated contract tests.

File: schemas/internal_server.proto [hunk 2]
```diff
@@ -85,9 +85,9 @@ message BookmarkPb { // Used for paginated responses
...
+        optional google.protobuf.Timestamp latest_create_time = 2;
+        string key_prefix = 3;
```
Comment (MateoRojas) [inline schemas/internal_server.proto:90]: What does key_prefix stand for? Maybe attribute string?

File: server/src/main/java/io/littlehorse/server/streamsimpl/lhinternalscan/publicrequests/SearchWfRun.java [hunk 3]
```diff
@@ -124,55 +124,46 @@ private List<String> searchAttributes() {
...
         }
         if (getLatestStart() != null) {
             tagScanBuilder.setLatestCreateTime(getLatestStart());
         }
+        tagScanBuilder.setKeyPrefix(getSearchAttributeString());
 
         out.setTagScan(tagScanBuilder.build());
         out.setType(ScanBoundaryCase.TAG_SCAN);
-        new WfRun()
+        TagStorageTypePb storageTypePb = new WfRun()
```
Comment (MateoRojas) [inline server/src/main/java/io/littlehorse/server/streamsimpl/lhinternalscan/publicrequests/SearchWfRun.java:142]: Is it better to call it `indexTypeForSearch`?

File: server/src/main/java/io/littlehorse/server/streamsimpl/lhinternalscan/publicrequests/SearchWfRun.java [hunk 4]
```diff
@@ -124,55 +124,46 @@ private List<String> searchAttributes() {
...
-                    out.setPartitionKey(
-                        Tag.getAttributeString(
-                            Getable.getTypeEnum(WfRun.class),
-                            buildTagAttributes()
-                        )
-                    );
-                }
-            });
+            .orElse(null);
+        if (storageTypePb != null) {
```
Comment (MateoRojas) [inline server/src/main/java/io/littlehorse/server/streamsimpl/lhinternalscan/publicrequests/SearchWfRun.java:155]: What happens if it's null?

**Review by MateoRojas — APPROVED** [review]
Reviewed with @coltmcnealy-lh :)

Good to go with just some minor comments and questions

File: schemas/internal_server.proto (hunk 2 above)
Comment (eduwercamacaro) [reply schemas/internal_server.proto:90]: Key prefix is the RocksDb key prefix that will be used by `BackendInternalCommons#doScan` to perform the scan operation on  the database.

File: server/src/main/java/io/littlehorse/server/streamsimpl/lhinternalscan/publicrequests/SearchWfRun.java (hunk 4 above)
Comment (eduwercamacaro) [reply server/src/main/java/io/littlehorse/server/streamsimpl/lhinternalscan/publicrequests/SearchWfRun.java:155]: Addressed in other branch

**Pull request merged**